api/intel/openamt/openamt.go

```diff
@@ -0,0 +1,152 @@
+package openamt
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+const (
+	MpsServerAddress          = "127.0.0.1" // TODO: this is the same IP that must be used for the OpenAMT stack deployment, is this value right?
+	MpsServerAdminUser        = "admin"
+	DefaultCIRAConfigName     = "ciraConfigDefault"
+	DefaultWirelessConfigName = "wirelessProfileDefault"
+	DefaultProfileName        = "profileAMTDefault"
+	DefaultDomainName         = "domainDefault"
+)
+
+// Service represents a service for managing an OpenAMT server.
+type Service struct {
+	httpsClient *http.Client
+}
+
+// NewService initializes a new service.
+func NewService() *Service {
+	return &Service{
+		httpsClient:
+		&http.Client{
+			Timeout: time.Second * time.Duration(5),
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			},
+		},
+	}
+}
+
+type openAMTError struct {
+	ErrorMsg string `json:"message"`
+	Errors   []struct {
+		ErrorMsg string `json:"msg"`
+	} `json:"errors"`
+}
+
+func parseError(responseBody []byte) error {
+	var errorResponse openAMTError
+	err := json.Unmarshal(responseBody, &errorResponse)
+	if err != nil {
+		return err
+	}
+
+	if len(errorResponse.Errors) > 0 {
+		return errors.New(errorResponse.Errors[0].ErrorMsg)
+	}
+	if errorResponse.ErrorMsg != "" {
+		return errors.New(errorResponse.ErrorMsg)
+	}
+	return nil
+}
+
+func (service *Service) ConfigureDefault(certFileText string, certPassword string, domainSuffix string, useWirelessConfig bool, wifiAuthenticationMethod int, wifiEncryptionMethod int, wifiSSID string, wifiPskPass string) error {
+	token, err := service.executeAuthenticationRequest()
+	if err != nil {
+		return err
+	}
+
+	ciraConfig, err := service.createOrUpdateCIRAConfig(token.Token, DefaultCIRAConfigName)
+	if err != nil {
+		return err
+	}
+
+	wirelessConfigName := ""
+	if useWirelessConfig {
+		wirlessConfig, err := service.createOrUpdateWirelessConfig(token.Token, DefaultWirelessConfigName, wifiAuthenticationMethod, wifiEncryptionMethod, wifiSSID, wifiPskPass)
+		if err != nil {
+			return err
+		}
+		wirelessConfigName = wirlessConfig.ProfileName
+	}
+
+	profile, err := service.createOrUpdateAMTProfile(token.Token, DefaultProfileName, ciraConfig.ConfigName, wirelessConfigName)
+	if err != nil {
+		return err
+	}
+
+	domain, err := service.createOrUpdateDomain(token.Token, DefaultDomainName, domainSuffix, certFileText, certPassword)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println(ciraConfig.ConfigName)
```

Comment:
use logrus.Infof() initially, and later we'll convert some of them to logrus.Debugf (we need a UI/api way to change log levels)

Reply:
Oops I meant to delete this "logs"